scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,279 @@
+                    npcUtil.tradeHasExactly(trade, xi.items.ELVAAN_F_MANNEQUIN) or
+                    npcUtil.tradeHasExactly(trade, xi.items.TARUTARU_M_MANNEQUIN) or
+                    npcUtil.tradeHasExactly(trade, xi.items.TARUTARU_F_MANNEQUIN) or
+                    npcUtil.tradeHasExactly(trade, xi.items.MITHRA_MANNEQUIN) or
+                    npcUtil.tradeHasExactly(trade, xi.items.GALKA_MANNEQUIN) then
+                        return quest:progressEvent(319, { [0] = 2,
+                                                          [1] = getMannequins(player), -- Player Mannequin List
+                                                          [2] = xi.mannequin.cost.PURCHASE,
+                                                          [3] = xi.mannequin.cost.TRADE,
+                                                          [4] = 1, -- Leaving this out gives a "It's still in pretty good condition" message
+                                                          })
+                    end
+                end,
+            },
+
+            onEventUpdate =
+            {
+                [321] = function(player, csid, option, npc)
+                    player:PrintToPlayer(string.format("Update 321: Option: %s", option))
+                    if option == 0 then
+                        -- Purchase a mannequin
+                        local richEnough = 0
+                        if player:getGil() >= xi.mannequin.cost.PURCHASE then richEnough = 1 end
+                        player:updateEvent( { [0] = richEnough, -- Not sure if this is the legitimate use, but it works.
+                                              [1] = getMannequins(player),
+                                              [2] = option,
+                                              })
+                    elseif option >= 11 and option <= 18 then
+                        -- Pose a mannequin
+                        local race = option - 10 -- From 1 to 8, for consistency in lua
+                        player:updateEvent( { [0] = 1,
+                                              [1] = getMannequins(player),
+                                              [2] = option,
+                                              [3] = getMannequinPose(player, race),
+                                              })
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [319] = function(player, csid, option, npc)
+                    -- Trade mannequin.  Option = race (1-8)
+                    print(option)
+                    if option >= 1 and option <= 8 and
+                    player:delGil(xi.mannequin.cost.TRADE) then
+                        player:tradeComplete()
+                        player:messageSpecial(ID.text.ITEM_OBTAINED, xi.items.HUME_M_MANNEQUIN + option - 1)
+                        player:addItem(xi.items.HUME_M_MANNEQUIN + option - 1)
```

Comment:
Adding the following should also give player the message they received the new mannequin. Unless that's not proper behavior
```suggestion
                        npcUtil.giveItem(xi.items.HUME_M_MANNEQUIN + option - 1)
```